app/src/main/java/org/kiwix/kiwixmobile/KiwixErrorActivity.java

```diff
@@ -0,0 +1,161 @@
+        Throwable exception = (Throwable) extras.getSerializable("exception");
+
+        reportButton.setOnClickListener(v -> {
+
+
+            Intent emailIntent = new Intent(Intent.ACTION_SEND);
+            emailIntent.setType("vnd.android.cursor.dir/email");
+            String to[] = {"android-crash-feedback@kiwix.org"};
+            emailIntent.putExtra(Intent.EXTRA_EMAIL, to);
+            emailIntent.putExtra(Intent.EXTRA_SUBJECT, "Someone has reported a crash");
+
+            String body = "Hi Kiwix Developers!\n" +
+                    "The Android app crashed, here are some details to help fix it:\n\n";
+
+            if(allowLogsCheckbox.isChecked()) {
+                File appDirectory = new File(Environment.getExternalStorageDirectory() + "/Kiwix");
+                File logFile = new File(appDirectory, "logcat.txt");
+                Uri path = Uri.fromFile(logFile);
+                emailIntent.putExtra(Intent.EXTRA_STREAM, path);
+            }
+
+            if(allowCrashCheckbox.isChecked()) {
+                body += "Exception Details:\n\n" +
+                        exception.toString() +
+                "\n\n";
+            }
+
+            if(allowZimsCheckbox.isChecked()) {
+                BookDao bookDao = new BookDao(KiwixDatabase.getInstance(getApplicationContext()));
+                ArrayList<LibraryNetworkEntity.Book> books = bookDao.getBooks();
+
+                String allZimFiles = "";
+                for(LibraryNetworkEntity.Book book: books) {
+                    String bookString = book.getTitle() +
+                            ":\nArticles: ["+ book.getArticleCount() +
+                            "]\nCreator: [" + book.getCreator() +
+                            "]\n";
+
+                    allZimFiles += bookString;
+                }
+
+                String currentZimFile = ZimContentProvider.getZimFile();
+                body += "Curent Zim File:\n" +
+                        currentZimFile +
+                        "\n\nAll Zim Files in DB:\n" +
+                        allZimFiles +
+                "\n\n";
+            }
+
+            if(allowLanguageCheckbox.isChecked()) {
+                body += "Current Locale:\n" +
+                        getCurrentLocale(getApplicationContext()) +
+                "\n\n";
+            }
+
+            if(allowDeviceDetailsCheckbox.isChecked()) {
+                body += "Device Details:\n" +
+                        "Device:[" + Build.DEVICE
+                        + "]\nModel:[" + Build.MODEL
+                        + "]\nManufacturer:[" + Build.MANUFACTURER
+                        + "]\nTime:[" + Build.TIME
+                        + "]\nAndroid Version:[" + Build.VERSION.RELEASE
+                        + "]" +
+                        "\n\n";
+            }
+
+            emailIntent.putExtra(Intent.EXTRA_TEXT, body);
+
+            startActivityForResult(Intent.createChooser(emailIntent , "Send email..."), 1);
```

Comment:
```FATAL EXCEPTION: main
android.os.FileUriExposedException: file:///storage/emulated/0/Kiwix/logcat.txt exposed beyond app through ClipData.Item.getUri()
 at android.os.StrictMode.onFileUriExposed(StrictMode.java:1958)
 at android.net.Uri.checkFileUriExposed(Uri.java:2348)
 at android.content.ClipData.prepareToLeaveProcess(ClipData.java:941)
 at android.content.Intent.prepareToLeaveProcess(Intent.java:9735)
 at android.content.Intent.prepareToLeaveProcess(Intent.java:9741)
 at android.content.Intent.prepareToLeaveProcess(Intent.java:9720)
 at android.app.Instrumentation.execStartActivity(Instrumentation.java:1609)
 at android.app.Activity.startActivityForResult(Activity.java:4472)
 at android.support.v4.app.BaseFragmentActivityApi16.startActivityForResult(BaseFragmentActivityApi16.java:54)
 at android.support.v4.app.FragmentActivity.startActivityForResult(FragmentActivity.java:67)
 at android.app.Activity.startActivityForResult(Activity.java:4430)
 at android.support.v4.app.FragmentActivity.startActivityForResult(FragmentActivity.java:732)
 at org.kiwix.kiwixmobile.KiwixErrorActivity.lambda$onCreate$0$KiwixErrorActivity(KiwixErrorActivity.java:133)
 at org.kiwix.kiwixmobile.KiwixErrorActivity$$Lambda$0.onClick(Unknown Source:23)
 at android.view.View.performClick(View.java:6256)
 at android.view.View$PerformClick.run(View.java:24701)
 at android.os.Handler.handleCallback(Handler.java:789)
 at android.os.Handler.dispatchMessage(Handler.java:98)
 at android.os.Looper.loop(Looper.java:164)
 at android.app.ActivityThread.main(ActivityThread.java:6541)
 at java.lang.reflect.Method.invoke(Native Method)
 at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)

Reply:
@EladKeyshawn Any thoughts?

Reply:
Where did this exception come from?

Reply:
@ISNIT0 I reckon its something to do with wrong file handling with the current target api, as it says here:
 https://stackoverflow.com/questions/38200282/android-os-fileuriexposedexception-file-storage-emulated-0-test-txt-exposed

Reply:
@mhutti1 Where did the exception come from? When you tested the app?

Reply:
I get this when attempting to report after a crash. I think the issue is you are attempting to pass a uri to a  log file via an intent but this is illegal in later version of Android. You are right that you should use a file provider or come up with another workaround.

Reply:
What a bore. Not sure I'll have much time to look at this - I'll try and set aside some time by Sunday

Reply:
A quick solution might be to pipe it as text. Although I am not sure what the limit on that is.